### ISSUE TITLE
CNN keras model started only giving NaN after change to keras-core

### ISSUE THREAD

**Issue by JotaFan** [keras-team/keras#18594]
I was using keras to build a simple CNN model. Input->batchnorm->conv->conv->droupout->dense->dense

I had done a lot of experiments and all good. I decided to change my builder from keras to keras-core, in order to be able to change backends, but in doing so I am constantly getting nan value in the loss now. And since I have an early stop for nan loss, I dont even get past the 1st epoch.

I it is working fine the same, mode, same data, with keras. but always goes to nan with keras-core.

Any ideas on this?

Edit: I have tried to tweek around optimizer, learning rate, kernel_initializer, activation, and still NaN. Increasing batch size did solve the issue, but I am still baffled as why it worked fine on keras and not on keras-core

**Comment by AakashKumarNain** [comment]
Can you provide a minimal example, a colab link?

**Comment by JotaFan** [comment]
Sorry for the delay.

So I can give you the json description of the models:

old tensorflow/keras working model:
`{"class_name": "Functional", "config": {"name": "model", "trainable": true, "layers": [{"module": "keras.layers", "class_name": "InputLayer", "config": {"batch_input_shape": [null, 168, 18], "dtype": "float32", "sparse": false, "ragged": false, "name": "input_1"}, "registered_name": null, "name": "input_1", "inbound_nodes": []}, {"module": "keras.layers", "class_name": "BatchNormalization", "config": {"name": "batch_normalization", "trainable": true, "dtype": "float32", "axis": [2], "momentum": 0.99, "epsilon": 0.001, "center": true, "scale": true, "beta_initializer": {"module": "keras.initializers", "class_name": "Zeros", "config": {}, "registered_name": null}, "gamma_initializer": {"module": "keras.initializers", "class_name": "Ones", "config": {}, "registered_name": null}, "moving_mean_initializer": {"module": "keras.initializers", "class_name": "Zeros", "config": {}, "registered_name": null}, "moving_variance_initializer": {"module": "keras.initializers", "class_name": "Ones", "config": {}, "registered_name": null}, "beta_regularizer": null, "gamma_regularizer": null, "beta_constraint": null, "gamma_constraint": null}, "registered_name": null, "build_config": {"input_shape": [null, 168, 18]}, "name": "batch_normalization", "inbound_nodes": [[["input_1", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Conv1D", "config": {"name": "conv1d", "trainable": true, "dtype": "float32", "filters": 16, "kernel_size": [3], "strides": [1], "padding": "valid", "data_format": "channels_last", "dilation_rate": [1], "groups": 1, "activation": "relu", "use_bias": true, "kernel_initializer": {"module": "keras.initializers", "class_name": "GlorotUniform", "config": {"seed": null}, "registered_name": null}, "bias_initializer": {"module": "keras.initializers", "class_name": "Zeros", "config": {}, "registered_name": null}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": null, "build_config": {"input_shape": [null, 168, 18]}, "name": "conv1d", "inbound_nodes": [[["batch_normalization", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "MaxPooling1D", "config": {"name": "max_pooling1d", "trainable": true, "dtype": "float32", "strides": [2], "pool_size": [2], "padding": "valid", "data_format": "channels_last"}, "registered_name": null, "build_config": {"input_shape": [null, 166, 16]}, "name": "max_pooling1d", "inbound_nodes": [[["conv1d", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Dropout", "config": {"name": "dropout", "trainable": true, "dtype": "float32", "rate": 0.35, "noise_shape": null, "seed": null}, "registered_name": null, "build_config": {"input_shape": [null, 83, 16]}, "name": "dropout", "inbound_nodes": [[["max_pooling1d", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Flatten", "config": {"name": "flatten", "trainable": true, "dtype": "float32", "data_format": "channels_last"}, "registered_name": null, "build_config": {"input_shape": [null, 83, 16]}, "name": "flatten", "inbound_nodes": [[["dropout", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Dense", "config": {"name": "dense", "trainable": true, "dtype": "float32", "units": 200, "activation": "linear", "use_bias": true, "kernel_initializer": {"module": "keras.initializers", "class_name": "GlorotUniform", "config": {"seed": null}, "registered_name": null}, "bias_initializer": {"module": "keras.initializers", "class_name": "Zeros", "config": {}, "registered_name": null}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": null, "build_config": {"input_shape": [null, 1328]}, "name": "dense", "inbound_nodes": [[["flatten", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Dense", "config": {"name": "dense_1", "trainable": true, "dtype": "float32", "units": 24, "activation": "linear", "use_bias": true, "kernel_initializer": {"module": "keras.initializers", "class_name": "GlorotUniform", "config": {"seed": null}, "registered_name": null}, "bias_initializer": {"module": "keras.initializers", "class_name": "Zeros", "config": {}, "registered_name": null}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": null, "build_config": {"input_shape": [null, 200]}, "name": "dense_1", "inbound_nodes": [[["dense", 0, 0, {}]]]}, {"module": "keras.layers", "class_name": "Reshape", "config": {"name": "reshape", "trainable": true, "dtype": "float32", "target_shape": [24, 1]}, "registered_name": null, "build_config": {"input_shape": [null, 24]}, "name": "reshape", "inbound_nodes": [[["dense_1", 0, 0, {}]]]}], "input_layers": [["input_1", 0, 0]], "output_layers": [["reshape", 0, 0]]}, "keras_version": "2.13.1", "backend": "tensorflow"}`

And the new keras-core version:
`
{"module": "keras_core.src.models.functional", "class_name": "Functional", "config": {"name": "functional_5", "trainable": true, "layers": [{"module": "keras_core.src.layers.core.input_layer", "class_name": "InputLayer", "config": {"batch_shape": [null, 168, 18], "dtype": "float32", "sparse": false, "name": "input_layer_2"}, "registered_name": "InputLayer", "name": "input_layer_2", "inbound_nodes": []}, {"module": "keras_core.src.layers.normalization.batch_normalization", "class_name": "BatchNormalization", "config": {"name": "batch_normalization_2", "trainable": true, "dtype": "float32", "axis": -1, "momentum": 0.99, "epsilon": 0.001, "center": true, "scale": true, "beta_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Zeros", "config": {}, "registered_name": "Zeros"}, "gamma_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Ones", "config": {}, "registered_name": "Ones"}, "moving_mean_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Zeros", "config": {}, "registered_name": "Zeros"}, "moving_variance_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Ones", "config": {}, "registered_name": "Ones"}, "beta_regularizer": null, "gamma_regularizer": null, "beta_constraint": null, "gamma_constraint": null}, "registered_name": "BatchNormalization", "build_config": {"input_shape": [null, 168, 18]}, "name": "batch_normalization_2", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 168, 18], "dtype": "float32", "keras_history": ["input_layer_2", 0, 0]}}], "kwargs": {"mask": null}}]}, {"module": "keras_core.src.layers.convolutional.conv1d", "class_name": "Conv1D", "config": {"name": "conv1d", "trainable": true, "dtype": "float32", "filters": 16, "kernel_size": [3], "strides": [1], "padding": "valid", "data_format": "channels_last", "dilation_rate": [1], "groups": 1, "activation": "relu", "use_bias": true, "kernel_initializer": {"module": "keras_core.src.initializers.random_initializers", "class_name": "GlorotUniform", "config": {"seed": null}, "registered_name": "GlorotUniform"}, "bias_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Zeros", "config": {}, "registered_name": "Zeros"}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": "Conv1D", "build_config": {"input_shape": [null, 168, 18]}, "name": "conv1d", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 168, 18], "dtype": "float32", "keras_history": ["batch_normalization_2", 0, 0]}}], "kwargs": {}}]}, {"module": "keras_core.src.layers.pooling.max_pooling1d", "class_name": "MaxPooling1D", "config": {"name": "max_pooling1d", "trainable": true, "dtype": "float32", "pool_size": [2], "padding": "valid", "strides": [2], "data_format": "channels_last"}, "registered_name": "MaxPooling1D", "build_config": {"input_shape": [null, 166, 16]}, "name": "max_pooling1d", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 166, 16], "dtype": "float32", "keras_history": ["conv1d", 0, 0]}}], "kwargs": {}}]}, {"module": "keras_core.src.layers.regularization.dropout", "class_name": "Dropout", "config": {"name": "dropout_1", "trainable": true, "dtype": "float32", "rate": 0.35, "seed": null, "noise_shape": null}, "registered_name": "Dropout", "name": "dropout_1", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 83, 16], "dtype": "float32", "keras_history": ["max_pooling1d", 0, 0]}}], "kwargs": {"training": false}}]}, {"module": "keras_core.src.layers.reshaping.flatten", "class_name": "Flatten", "config": {"name": "flatten_1", "trainable": true, "dtype": "float32", "data_format": "channels_last"}, "registered_name": "Flatten", "build_config": {"input_shape": [null, 83, 16]}, "name": "flatten_1", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 83, 16], "dtype": "float32", "keras_history": ["dropout_1", 0, 0]}}], "kwargs": {}}]}, {"module": "keras_core.src.layers.core.dense", "class_name": "Dense", "config": {"name": "dense_3", "trainable": true, "dtype": "float32", "units": 200, "activation": "relu", "use_bias": true, "kernel_initializer": {"module": "keras_core.src.initializers.random_initializers", "class_name": "HeNormal", "config": {"seed": null}, "registered_name": "HeNormal"}, "bias_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Zeros", "config": {}, "registered_name": "Zeros"}, "kernel_regularizer": null, "bias_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": "Dense", "build_config": {"input_shape": [null, 1328]}, "name": "dense_3", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 1328], "dtype": "float32", "keras_history": ["flatten_1", 0, 0]}}], "kwargs": {}}]}, {"module": "keras_core.src.layers.core.dense", "class_name": "Dense", "config": {"name": "dense_4", "trainable": true, "dtype": "float32", "units": 24, "activation": "relu", "use_bias": true, "kernel_initializer": {"module": "keras_core.src.initializers.random_initializers", "class_name": "HeNormal", "config": {"seed": null}, "registered_name": "HeNormal"}, "bias_initializer": {"module": "keras_core.src.initializers.constant_initializers", "class_name": "Zeros", "config": {}, "registered_name": "Zeros"}, "kernel_regularizer": null, "bias_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "registered_name": "Dense", "build_config": {"input_shape": [null, 200]}, "name": "dense_4", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 200], "dtype": "float32", "keras_history": ["dense_3", 0, 0]}}], "kwargs": {}}]}, {"module": "keras_core.src.layers.reshaping.reshape", "class_name": "Reshape", "config": {"name": "reshape_2", "trainable": true, "dtype": "float32", "target_shape": [24, 1]}, "registered_name": "Reshape", "build_config": {"input_shape": [null, 24]}, "name": "reshape_2", "inbound_nodes": [{"args": [{"class_name": "__keras_tensor__", "config": {"shape": [null, 24], "dtype": "float32", "keras_history": ["dense_4", 0, 0]}}], "kwargs": {}}]}], "input_layers": [["input_layer_2", 0, 0]], "output_layers": [["reshape_2", 0, 0]]}, "registered_name": "Functional", "build_config": {"input_shape": null}}`

**Comment by AakashKumarNain** [comment]
Can you please provide a simple working code instead? That way it would be much easy for us to provide the solution

**Comment by JotaFan** [comment]
I am using a custom builder, so anything that comes from here:
https://github.com/alquimodelia/forecat/blob/main/forecat/archs.py

And the difference was the imports, where before this issue the imports instead `from keras_core`, they were just `from keras`.

Something that "solved" was increasing size_batch for one, and running keras_core with JAX backend for another

**Comment by SuryanarayanaY** [comment]
Hi @JotaFan ,

This might be due to an issue with `batch_normalization` layer that got fixed subsequently in current keras_nightly.   Could you please test with keras_nightly and let us know?

Thanks!

**Comment by JotaFan** [comment]
Hello @SuryanarayanaY, thanks for the reply

I have updated to keras 3, and since then I never saw this issue again.

I apologize for not coming back here to report it!

**Comment by SuryanarayanaY** [comment]
Hi @JotaFan ,

Could you please mark the issue as resolved now? Thanks!
